user-console/src/main/java/org/pentaho/mantle/client/admin/EmailAdminPanelController.java

```diff
@@ -156,18 +151,41 @@ public void onValueChange( final ValueChangeEvent<Boolean> booleanValueChangeEve
 
     protocolsListBox.addChangeHandler( new ChangeHandler() {
       public void onChange( final ChangeEvent changeEvent ) {
-        emailConfig.setSmtpProtocol( protocolsListBox.getItemText( protocolsListBox.getSelectedIndex() ) );
+        String protocol = protocolsListBox.getValue( protocolsListBox.getSelectedIndex() );
+        smtpPanel.setVisible( protocol.contains( "SMTP" ) );
+        useSSLCheckBox.setVisible( protocol.contains( "SMTP" ) );
+        useStartTLSCheckBox.setVisible( protocol.contains( "SMTP" ) );
+        if ( protocol.equalsIgnoreCase( "graph_api" ) ) {
+          authenticationTypeLB.setSelectedIndex( 0 );
+          emailConfig.setAuthMechanism( authenticationTypeLB.getValue( 0 ) );
+          emailConfig.setAuthenticate( true );
+          authenticationTypeLB.setEnabled( false );
+          oauthPanel.setVisible( true );
+          passwordPanel.setVisible( false );
+          authenticationPanel.setVisible( true );
+        } else {
+          authenticationTypeLB.setEnabled( true );
+        }
+        emailConfig.setSmtpProtocol( protocol );
         setDirty( true );
       }
     } );
 
     saveButton.addClickHandler( new ClickHandler() {
       public void onClick( final ClickEvent clickEvent ) {
         if ( isPortValid( portTextBox.getValue() ) ) {
-          setEmailConfig();
+          String protocol = protocolsListBox.getValue( protocolsListBox.getSelectedIndex() );
+          String smtpHost =  smtpHostTextBox.getValue();
+          String authMechanism = authenticationTypeLB.getValue( authenticationTypeLB.getSelectedIndex() );
+          if ( isSmtpMicrosoftOAuth( protocol, smtpHost, authMechanism ) ) {
+            new MessageDialogBox( Messages.getString( ERROR ), Messages.getString( "oauthSupportError" ),
```

Comment:
For consistency, should create a constant for `"oauthSupportError"` as well.

Reply:
I concur see https://github.com/pentaho/pentaho-platform/pull/5185/files#r1081881270